swap_meet/vendor.py

```diff
@@ -1,2 +1,122 @@
+from operator import contains, truediv
+from swap_meet.item import Item
+
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory = None):
+        if not inventory:
+            inventory = list()
+        self.inventory = inventory
+    
+    def add(self, item):
+        """
+        Returns the item that is added to the inventory.
+        """
+        self.inventory.append(item)
+        return item
+    
+    def remove(self, item):
+        """
+        Returns the matching item and removes it from the inventory.
+        If there is no matching item, return False.
+        """
+        try:
+            self.inventory.remove(item)
+            return item
+        except ValueError:
+            return False
+
+    def get_by_category(self, category):
+        """
+        Returns a list of Items in the inventory with given category.
+        """
+        item_list = [item for item in self.inventory if item.category == category]
+        return item_list
+    
+    def swap_items(self, other, my_item, their_item):
+        """
+        Swaps this Vendor's my_item with the other Vendor's their_item. 
+        Returns True if swap is successful.
+        Returns False, if this Vendor's inventory doesn't contain my_item
+        or the other Vendor's inventory doesn't contain their_item.
+        """
+        if my_item in self.inventory and their_item in other.inventory:
+            return self.swap(other, my_item, their_item)
```

Comment:
Since `swap_items` is already supposed to do swapping of items, it doesn't really make sense to create a helper function to do this functionality. This function is already kind of the helper function that the other functions like `swap_first_item` should call. It makes things messy to have a helper function named so similar that does what this function is supposed to do.